### PR TITLE
Decouple Presence from Avatar View

### PR DESCRIPTION
**Pull request by andela-wakinsanmi**

### CODE REVIEW

File: layer-atlas/src/androidTest/java/com/layer/ui/test/AvatarViewTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Test
+    public void testThatAvatarColorChangeWhenSpinnerIsChanged() {
+        String selectionText = "AWAY";
+
+        onView(withId(R.id.test_spinner)).perform(click());
+        onData(allOf(is(instanceOf(String.class)), is(selectionText))).perform(click());
+        onView(withId(R.id.test_spinner)).check(matches(withSpinnerText(containsString(selectionText))));
+        onView(withText(selectionText)).perform(click());
+        //onView(withId(R.id.test_presence)).check(matches(withBgColor(mAwayColor)));
```
Comment (thecombatwombat) [inline layer-atlas/src/androidTest/java/com/layer/ui/test/AvatarViewTest.java]: remove if not needed

File: circle.yml [hunk 2]
```diff
@@ -9,10 +9,15 @@ test:
         parallel: true
     # wait for it to have booted
     - circle-android wait-for-boot
+
+# Unlock the emulator screen after 15 seconds
+    - sleep 45
```
Comment (sriamar) [inline circle.yml]: that is 45 sec?

File: layer-atlas/src/main/java/com/layer/ui/presence/PresenceView.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import java.util.Collection;
+import java.util.Set;
+
+public class PresenceView extends View {
+
+    private final Paint mBackgroundPaint = new Paint();
+    private final Paint mPresencePaint = new Paint();
+
+    private Identity mIdentity;
+    private int mAvailableColor = Color.rgb(0x4F, 0xBF, 0x62);
```
Comment (sriamar) [inline layer-atlas/src/main/java/com/layer/ui/presence/PresenceView.java]: All these colors should be pulled in from theme file?

File: layer-atlas/src/main/java/com/layer/ui/presence/PresenceView.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                drawOffline(canvas);
+                break;
+            case INVISIBLE:
+                drawInvisible(canvas);
+                break;
+            case BUSY:
+                drawBusy(canvas);
+                break;
+            default:
+                drawDefault(canvas);
```
Comment (sriamar) [inline layer-atlas/src/main/java/com/layer/ui/presence/PresenceView.java]: I don't understand this drawDefault. This switch statement is different from the one before in AvatarView().

File: layer-atlas/src/main/java/com/layer/ui/presence/PresenceView.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        canvas.drawCircle(presenceCenterX, presenceCenterY, presenceInnerRadius, mPresencePaint);
+
+        // Draw hollow if needed
+        if (makeCircleHollow) {
+            canvas.drawCircle(presenceCenterX, presenceCenterY, (presenceInnerRadius / 2f), mBackgroundPaint);
+        }
+    }
+
+    @VisibleForTesting
+    public int getPresenceColor() {
```
Comment (sriamar) [inline layer-atlas/src/main/java/com/layer/ui/presence/PresenceView.java]: This should be package private? i.e., leave out `public` ?

File: layer-atlas/src/main/java/com/layer/ui/presence/PresenceView.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+    public void setParticipants(Set<Identity> participants) {
+        showPresenceIfIdentityIsOne(participants);
+    }
+
+    public void setParticipants(Identity... participants) {
+        showPresenceIfIdentityIsOne(Arrays.asList(participants));
+    }
+
+    private void showPresenceIfIdentityIsOne(Collection<Identity> participants) {
```
Comment (sriamar) [inline layer-atlas/src/main/java/com/layer/ui/presence/PresenceView.java]: This method is not factored right. Here is all the things it does.
- if participant size is 1, it stores participant value in mIdentity. 
- Shows / hides presence
- requests redraw.
The name doesn't convey that. 


You can make the method name generic like `processSetParticipants()` instead?

**Comment by sriamar** [comment]
And, your test failed! Let's fix that too :)

**Pull request merged**